teamscale-client/src/main/java/com/teamscale/client/ProxySystemProperties.java

```diff
@@ -51,120 +58,100 @@ public ProxySystemProperties(Protocol protocol) {
-		return protocol + PROXY_HOST_SYSTEM_PROPERTY;
+	protected String getProxyHostSystemPropertyName() {
+		return getPropertyPrefix() + protocol + PROXY_HOST_SYSTEM_PROPERTY;
 	}
 
-	/**
-	 * Set the http(s).proxyPort system variable
-	 */
+	/** Set the http(s).proxyPort system variable. */
 	public void setProxyPort(int proxyPort) {
 		setProxyPort(proxyPort + "");
 	}
 
-	/**
-	 * Set the http(s).proxyPort system variable
-	 */
+	/** Set the http(s).proxyPort system variable. */
 	public void setProxyPort(String proxyPort) {
 		System.setProperty(getProxyPortSystemPropertyName(), proxyPort);
 	}
 
-	/**
-	 * Removes the http(s).proxyPort system variable.
-	 * For testing.
-	 */
+	/** Removes the http(s).proxyPort system variable. For testing. */
 	/*package*/ void removeProxyPort() {
 		System.clearProperty(getProxyPortSystemPropertyName());
 	}
 
+	/** @return the name of the system property specifying the proxy port. */
 	@NotNull
-	private String getProxyPortSystemPropertyName() {
-		return protocol + PROXY_PORT_SYSTEM_PROPERTY;
+	protected String getProxyPortSystemPropertyName() {
+		return getPropertyPrefix() + protocol + PROXY_PORT_SYSTEM_PROPERTY;
 	}
 
-	/**
-	 * Get the http(s).proxyUser system variable
-	 */
+	/** Get the http(s).proxyUser system variable. */
 	public String getProxyUser() {
 		return System.getProperty(getProxyUserSystemPropertyName());
 	}
 
-	/**
-	 * Set the http(s).proxyUser system variable
-	 */
+	/** Set the http(s).proxyUser system variable. */
 	public void setProxyUser(String proxyUser) {
 		System.setProperty(getProxyUserSystemPropertyName(), proxyUser);
 	}
 
+	/** @return the name of the system property specifying the proxy user. */
 	@NotNull
-	private String getProxyUserSystemPropertyName() {
-		return protocol + PROXY_USER_SYSTEM_PROPERTY;
+	protected String getProxyUserSystemPropertyName() {
+		return getPropertyPrefix() + protocol + PROXY_USER_SYSTEM_PROPERTY;
 	}
 
-	/**
-	 * Get the http(s).proxyPassword system variable
-	 */
+	/** Get the http(s).proxyPassword system variable. */
 	public String getProxyPassword() {
 		return System.getProperty(getProxyPasswordSystemPropertyName());
 	}
 
 
-	/**
-	 * Set the http(s).proxyPassword system variable
-	 */
+	/** Set the http(s).proxyPassword system variable. */
 	public void setProxyPassword(String proxyPassword) {
 		System.setProperty(getProxyPasswordSystemPropertyName(), proxyPassword);
 	}
 
+	/** @return the name of the system property specifying the proxy password. */
 	@NotNull
-	private String getProxyPasswordSystemPropertyName() {
-		return protocol + PROXY_PASSWORD_SYSTEM_PROPERTY;
+	protected String getProxyPasswordSystemPropertyName() {
+		return getPropertyPrefix() + protocol + PROXY_PASSWORD_SYSTEM_PROPERTY;
 	}
 
 	/** Parses the given port string. Returns -1 if the string is null or not a valid number. */
-	private int parsePort(String portString) {
+	private int parsePort(String portString, Consumer<String> logFunction) {
```

Comment:
To be honest I very much dislike this. The problem is that we need the `ProxySystemProperties`/`TeamscaleProxySystemProperties` in the `TeamscaleProxyOptions`, and at that point we would have to use a `DelayedLogger`, as it happens at runtime before our logging config has been read and the default one does not allow "shaded FQCNs". But later in time we want to be able to log to a real logger... Do you have any idea how to improve this? I'm open to having a brainstorming session.